### PR TITLE
Add proxy middleware

### PR DESCRIPTION
**Pull request by wllmwu**
## Changes

What changes did you make? Include screenshots if applicable, or explain how to view the changes.

- ~~Fix "Invalid options object" error on `npm start` by using `http-proxy-middleware` package instead of `proxy` line in `package.json`~~
- Fix "Invalid options object" error when starting the frontend on some machines by using environment variables and CORS headers
- Remove `proxy` line from `frontend/package.json`

## Testing

How did you confirm your changes work? (Automated tests, manual verification, etc.)

- ~~Not sure how this has been tested -@wllmwu~~
- Manually tested task creation on the frontend - new task appears in database as expected

## Tracking

Add your issue number below.

No issue was created for this bug, see **Changes** above

### CODE REVIEW

**Comment by wllmwu** [comment]
@navidboloorian Can you describe what you did to test this before I merge it?

**Comment by wllmwu** [comment]
> @navidboloorian Can you describe what you did to test this before I merge it?

Never mind, I implemented a fix which I think is more likely to work across systems. Test it yourself when you review.

File: frontend/src/api/requests.ts [hunk 1]
```diff
@@ -114,7 +121,7 @@ export async function put(
   body: unknown,
   headers: Record<string, string> = {},
 ): Promise<Response> {
-  const response = await fetchRequest("GET", url, body, headers);
+  const response = await fetchRequest("PUT", API_BASE_URL + url, body, headers);
```
Comment (wllmwu) [inline frontend/src/api/requests.ts:124]: @navidboloorian This was a little sloppy tbh, I should have looked more closely when reviewing

**Review by navidboloorian — APPROVED** [review]
works on my system

**Pull request merged**